Packet++/header/HttpLayer.h

```diff
@@ -207,201 +207,217 @@ namespace pcpp
 		HttpRequestFirstLine* m_FirstLine;
 	};
 
-
-
-
-
-	// -------- Class HttpResponseLayer -----------------
-
-	class HttpResponseFirstLine;
-
+	// -------- Class HttpResponseStatusCode -----------------
 
 	/**
-	 * @class HttpResponseLayer
-	 * Represents an HTTP response header and inherits all basic functionality of HttpMessage and TextBasedProtocolMessage.
-	 * The functionality that is added for this class is the HTTP first line concept. An HTTP response has the following first line:
-	 * <i>200 OK HTTP/1.1</i>
-	 * Since it's not an "ordinary" HTTP field, it requires a special treatment and gets a class of it's own: HttpResponseFirstLine.
-	 * Unlike most L2-4 protocols, an HTTP response header can spread over more than 1 packet. PcapPlusPlus currently doesn't support a header
-	 * that is spread over more than 1 packet so in such cases: 1) only the first packet will be parsed as HttpResponseLayer (the other packets
-	 * won't be recognized as HttpResponseLayer) and 2) the HTTP header for the first packet won't be complete (as it continues in the following
-	 * packets), this why PcapPlusPlus can indicate that HTTP response header is complete or not (doesn't end with "\r\n\r\n" or "\n\n") using
-	 * HttpMessage#isHeaderComplete()
+	 * Class for the enum of HTTP response status codes
 	 */
-	class HttpResponseLayer : public HttpMessage
+	class HttpResponseStatusCode
 	{
-		friend class HttpResponseFirstLine;
 	public:
-		/**
-		 * Enum for HTTP response status codes
-		 */
-		enum HttpResponseStatusCode
+		enum Value: int
```

Comment:
shouldn't this be an `enum class`?

Reply:
if we make it `enum class`, then we will need to call `HttpResponseStatusCode::Value::Http417ExpectationFailed`, I think it's okay to be `enum`, since the scope of the enum is already restricted in `HttpResponseStatusCode`

Reply:
yeah, it's valid.
The `enum Value : int` syntax specifies that the enumeration constants should have an underlying type of int, and that's why `static<int>(statusCode)` works

Reply:
ok, I didn't know that. In that case it's ok